ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/homescreenpage/HomeScreenContainerFragmentTest.java

```diff
@@ -0,0 +1,206 @@
+import com.kayako.sdk.android.k5.R;
+import com.kayako.sdk.android.k5.core.MessengerStylePref;
+import com.kayako.sdk.android.k5.messenger.style.MessengerTemplateHelper;
+import static org.hamcrest.CoreMatchers.is;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ErrorCollector;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.support.membermodification.MemberMatcher.methods;
+import static org.powermock.api.support.membermodification.MemberModifier.replace;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import static org.powermock.api.mockito.PowerMockito.suppress;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+
+@PrepareForTest({
+        MessengerStylePref.class,
+        MessengerTemplateHelper.class,
+        Fragment.class
+})
+@RunWith(PowerMockRunner.class)
+public class HomeScreenContainerFragmentTest {
+
+    private final HomeScreenContainerFragment homeScreenContainerFragment =
+            new HomeScreenContainerFragment();
+
+    @Mock
+    private Bundle bundle;
+
+    @Mock
+    private LayoutInflater layoutInflater;
+
+    @Mock
+    private ViewGroup viewGroup;
+
+    @Mock
+    private View view;
+
+    @Mock
+    private MessengerStylePref messengerStylePref;
+
+    @Mock
+    private HomeScreenContainerContract.Presenter presenter;
+
+    @Mock
+    private AppCompatActivity appCompatActivity;
+
+    @Mock
+    private HomeScreenListFragment homeScreenListFragment;
+
+    @Mock
+    private FragmentManager fragmentManager;
+
+    @Captor
+    private ArgumentCaptor<Integer> integerArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<Boolean> booleanArgumentCaptor;
+
+    @Rule
+    public final ErrorCollector errorCollector = new ErrorCollector();
+
+    @Before
+    public void setUp() {
+        mockStatic(MessengerStylePref.class);
+        when(MessengerStylePref.getInstance()).thenReturn(messengerStylePref);
+    }
+
+    @Test
+    public void onCreate() {
+        homeScreenContainerFragment.onCreate(bundle);
+        final HomeScreenContainerContract.Presenter mPresenter =
+                Whitebox.getInternalState(homeScreenContainerFragment, "mPresenter");
+        errorCollector.checkThat(mPresenter, notNullValue());
+    }
+
+    @Test
+    public void onCreateView() {
+        //Arrange
```

Comment:
✅ Good to see that the AAA("Arrange-Act-Assert") arranging and formatting pattern is followed.

Some of the main advantages are that it clearly separates what is being tested from the setup and verification steps. It also clarifies and focuses attention on a historically successful and generally necessary set of test steps.

**No change suggested, looks good!**	  

<a id="a41a1900-54f3-11e8-9db9-b7d507f482ce" href="none" codereview-uuid-reserved-tag />